src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -46,6 +46,88 @@ def boolified_type(symbol_type: ct.SymbolType) -> ct.ScalarType | ct.FieldType:
     raise GTTypeError(f"Can not boolify type {symbol_type}!")
 
 
+def extract_promoted_tuple_type(
+    node_type_ls: list[ct.TupleType],
+    mask_type: ct.FieldType,
+    promoted_type: None | ct.FieldType | ct.ScalarType,
+) -> ct.FieldType:
+    """
+    Extract argument type in tuple to which all others need to be promoted to.
+
+    If arguments types in a tuple are different (e.g. some are Fields and others Scalars), they all need to be promoted to be the same.
+    This function extract which type the promotion has to be done with respect of.
```

Comment:
```suggestion
    Extract argument type in tuple to which all others need to be promoted.

    If arguments types in a tuple are different (e.g. some are Fields and others Scalars), they all need to be promoted to be the same.
    This function extracts the type with respect to which the promotion has to be done.
```

In addition to the phrasing I am questioning the described semantics. Do all tuple elements necessarily have to be the same? What happens if the tuples contain mixed dimension fields? See comment in the type deduction tests for a better description of the question.

Reply:
If you mix dimensions, all items will be promoted to a mutual type. Such type contains all dimensions from all inout fields